vector/src/main/java/im/vector/app/features/home/HomeActivityViewModel.kt

```diff
@@ -117,17 +116,36 @@ class HomeActivityViewModel @AssistedInject constructor(
     private fun initialize() {
         if (isInitialized) return
         isInitialized = true
+        registerUnifiedPushIfNeeded()
         cleanupFiles()
         observeInitialSync()
         checkSessionPushIsOn()
         observeCrossSigningReset()
         observeAnalytics()
         observeReleaseNotes()
-        observeLocalNotificationsSilenced()
         initThreadsMigration()
         viewModelScope.launch { stopOngoingVoiceBroadcastUseCase.execute() }
     }
 
+    private fun registerUnifiedPushIfNeeded() {
+        if (vectorPreferences.areNotificationEnabledForDevice()) {
+            registerUnifiedPush(distributor = "")
+        }
+    }
+
+    private fun registerUnifiedPush(distributor: String) {
+        viewModelScope.launch {
+            when (registerUnifiedPushUseCase.execute(distributor = distributor)) {
+                is RegisterUnifiedPushUseCase.RegisterUnifiedPushResult.NeedToAskUserForDistributor -> {
+                    _viewEvents.post(HomeActivityViewEvents.AskUserForPushDistributor)
+                }
+                RegisterUnifiedPushUseCase.RegisterUnifiedPushResult.Success -> {
+                    ensureFcmTokenIsRetrievedUseCase.execute(pushersManager, registerPusher = vectorPreferences.areNotificationEnabledForDevice())
+                }
+            }
+        }
+    }
```

Comment:
Thanks for extracting logic from the Activity to the ViewModel. Maybe creating a new dedicated ViewModel may have been better, but it can be done later.